### ISSUE TITLE
updateauth not recorded by v1 history plugin if account does not exist when referenced

### ISSUE THREAD

**Issue by blockpane** [fioprotocol/fio#161]
It appears one of the history tables isn't getting updated when an account does not exist but is referenced in an updateauth call:

Example, two accounts both having been delegated a linked permission: one shows no trace in either the block explorer or the get_controlled_accounts endpoint and other does:

```
curl -s -XPOST http://testnet.fio.dev/v1/history/get_controlled_accounts -d '{
  "controlling_account": "gjkdwmtx2tse"
}'
...
{
  "controlled_accounts": [
    "l35p5swqgvoo"
  ]
}
```

vs.

```
curl -s -XPOST http://testnet.fio.dev/v1/history/get_controlled_accounts -d '{
  "controlling_account": "voterxbrq3vw"
}'
...
{
  "controlled_accounts": []
}
```

The difference is that the first account was created via newaccount after a trnsfiopubky. The second account was linked before the account had been created.

It may be a good idea to either 1) reject updateauth referencing a public key without an entry in the accountmap table for fio.address, or 2) modify the v1 history plugin to index the event. I'm guessing option 1 is much less work. 

Neither account show traces of the activity when calling get_actions, though I'm not certain this behavior differs from eos' v1 history plugin and I suspect that's ok - indexing that would require inspecting the action details since the account is neither the receiver or actor.

**Comment by tagdapix** [comment]
should have noted, the expected result for the second get_controlled_accounts api call should include `rssuh33ljdbm`

**Comment by adsorptionenthalpy** [comment]
We need to explicitly index the entry when an an account that was just created with newaccount is involved in subsequent inline actions, we're doing this for a few others already and I suspect there could be a few more. I'm already working in history.cpp at the moment so I will take a peek. 
Option 2 is more uniform with current logic and I think should be included, we should explore option 1 in more detail.

**Comment by tagdapix** [comment]
On second though, option 1 probably isn't desirable because when using updateauth (but only when exclusively using keys as the authority) the controlled account pays the fees for a linked action. There is some utility in this scenario, as it is only signature validation--an account isn't needed because the actor and permission are both tied to the controlled account. 

For example, the following sequence (assuming the pubkey in the updateauth has no existing account) works:
```
clio -u http://dev:8889 set account permission o2ouxipw2rt4 nonexist '{
    "threshold": 1,
    "keys": [
      {
        "key": "FIO8mu91eebrTHhesDw1fDf3MgPqDTVveFmynQYQkyqke1UZHftUU",
        "weight": 1
      }
    ],
    "accounts": [],
    "waits": []
  }
' 1000000000 active -p o2ouxipw2rt4

clio -u http://dev:8889 set action permission o2ouxipw2rt4 fio.address regaddress nonexist 400000000 -p o2ouxipw2rt4

clio wallet import --private-key 5JXMDmoU8..........................................

clio -u http://dev:8889 push action fio.address regaddress '{
  "fio_address": "shouldthisbeallowed@dapixdev",
  "owner_fio_public_key": "FIO6G9pXXM92Gy5eMwNquGULoCj3ZStwPLPdEb9mVXyEHqWN7HSuA",
  "max_fee": 40000000000,
  "actor": "o2ouxipw2rt4",
  "tpid": ""
}' -p o2ouxipw2rt4@nonexist
```

This isn't entirely different than using a signing key for block production, it's just a signature and there is no "accounting" involved for the key, hence no need for an account. 

If a client were to derive the pubkey from the signature, and assumed a 1:1 relationship between a public key and account--as is normally the case in FIO, it could cause some confusion, though I think the utility of using arbitrary keys with linkauth overrides this, and I think it's an edge case where a client go through the effort to derive the pubkey from the signature, then convert it to an account instead of looking at the authority provided in the transaction.

It is noteworthy that attempting updateauth using the accounts vector (ie msig) does fail when the account does not exist. 

TL;DR you are right, this is solely a history issue.

**Comment by ericbutz** [comment]
Added updateauth to chain history plugin so actions get added to the chain log as per the PR above. Closing.
